domain-service/credit-application/src/main/java/at/deloittedigital/core_banking/credit/datasource/RemoteCreditDataService.java

```diff
@@ -0,0 +1,49 @@
+package at.deloittedigital.core_banking.credit.datasource;
+
+import at.deloittedigital.core_banking.credit.datasource.model.CreditData;
+import at.deloittedigital.core_banking.credit.datasource.model.QueryResult;
+import java.net.URI;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+@Service
+@Log4j2
+@ConditionalOnProperty(name = "core-banking.credit.datasource.mode", havingValue = "remote")
+public class RemoteCreditDataService implements CreditDataService {
+
+    private final RestTemplate restTemplate;
+    private final URI dataServiceUrl;
+
+    public RemoteCreditDataService(
+            RestTemplate restTemplate,
+            @Value("${core-banking.credit.datasource.url}") URI dataServiceUrl) {
+        this.restTemplate = restTemplate;
+        this.dataServiceUrl = dataServiceUrl;
+    }
+
+    @Override
+    public CreditData query(String clientId) {
+        // for testing basic service integration
+        // todo error handling, non-blocking req, response validity check
```

Comment:
Are you planning to use the webflux web client for non-blocking requests? Maybe it makes sense to replace the `RestTemplate` already then. You can use the web client in a blocking fashion too and it might make refactoring later easier. Just an idea though.

Reply:
I was planning to use some graphql-specific client library, but I rather not integrate it for the first iteration, hence I decided just to do the simple call with `RestTemplate`. As long as we have just one test endpoint, switching it to any other lib, sync or async call should not be a serious effort.